lib/libzfs/os/freebsd/libzfs_zmount.c

```diff
@@ -120,16 +120,19 @@ do_mount_(const char *spec, const char *dir, int mflag, char *fstype,
 int
 do_mount(zfs_handle_t *zhp, const char *mntpt, char *opts, int flags)
 {
-
 	return (do_mount_(zfs_get_name(zhp), mntpt, flags, MNTTYPE_ZFS, NULL, 0,
 	    opts, sizeof (mntpt)));
 }
 
 int
 do_unmount(const char *mntpt, int flags)
 {
+	int rv;
 
-	return (unmount(mntpt, flags));
+	rv = unmount(mntpt, flags);
+	if (rv < 0)
+		return (errno);
```

Comment:
This could be simplified:

`if (unmount(mntpt, flags) < 0)
            return (errno);
return (0);`

Reply:
indeed! fixed